tools/esmf-aspect-model-maven-plugin/src/main/java/org/eclipse/esmf/aspectmodel/GenerateAsyncApiSpec.java

```diff
@@ -0,0 +1,128 @@
+   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+   private static final ObjectMapper YAML_MAPPER = new YAMLMapper().enable( YAMLGenerator.Feature.MINIMIZE_QUOTES );
+   private static final Logger LOG = LoggerFactory.getLogger( GenerateAsyncApiSpec.class );
+
+   private final AspectModelAsyncApiGenerator generator = new AspectModelAsyncApiGenerator();
+
+   @Parameter( required = true )
+   private String outputFormat = "";
+   @Parameter( defaultValue = "false" )
+   private boolean separateFiles;
+   @Parameter
+   private String applicationId;
+   @Parameter
+   private String channelAddress;
+   @Parameter( defaultValue = "false" )
+   private boolean useSemanticApiVersion;
+   @Parameter( defaultValue = "en" )
+   private String language;
+
+   @Override
+   public void execute() throws MojoExecutionException, MojoFailureException {
+      final Set<AspectContext> aspectModels = loadModelsOrFail();
+      final Locale locale = Optional.ofNullable( language ).map( Locale::forLanguageTag ).orElse( Locale.ENGLISH );
+      final ApiFormat format = Try.of( () -> ApiFormat.valueOf( outputFormat.toUpperCase() ) )
+            .getOrElseThrow( () -> new MojoExecutionException( "Invalid output format." ) );
+      for ( final AspectContext context : aspectModels ) {
+         final Aspect aspect = context.aspect();
+         final AsyncApiSchemaGenerationConfig config = AsyncApiSchemaGenerationConfigBuilder.builder()
+               .useSemanticVersion( useSemanticApiVersion )
+               .applicationId( applicationId )
+               .channelAddress( channelAddress )
+               .locale( locale )
+               .build();
+
+         final AsyncApiSchemaArtifact asyncApiSpec = generator.apply( aspect, config );
+         try {
+            if ( separateFiles ) {
+               writeSchemaWithSeparateFiles( format, asyncApiSpec, generator, aspect.getName() );
+            } else {
+               writeSchemaWithInOneFile( aspect.getName() + ".aai." + format.toString().toLowerCase(), format, asyncApiSpec );
+            }
+         } catch ( final IOException exception ) {
+            throw new MojoExecutionException( "Could not generate AsyncAPI specification.", exception );
+         }
+      }
+   }
+
+   private void writeSchemaWithInOneFile( final String schemaFileName, final ApiFormat format, final AsyncApiSchemaArtifact asyncApiSpec )
+         throws IOException {
+      try ( final OutputStream out = getOutputStreamForFile( schemaFileName, outputDirectory ) ) {
+         if ( format == ApiFormat.JSON ) {
+            OBJECT_MAPPER.writerWithDefaultPrettyPrinter().writeValue( out, asyncApiSpec.getContent() );
+         } else {
+            out.write( jsonToYaml( asyncApiSpec.getContent() ).getBytes( StandardCharsets.UTF_8 ) );
+         }
+      }
+   }
+
+   private void writeSchemaWithSeparateFiles( final ApiFormat format, final AsyncApiSchemaArtifact asyncApiSpec,
```

Comment:
I see a lot of redundancy here, would this be a way of resolving it?

```java
private void writeSchemaWithSeparateFiles(final ApiFormat format, final AsyncApiSchemaArtifact asyncApiSpec, final AspectModelAsyncApiGenerator generator, final String aspectName) throws IOException {
    final Path root = Path.of(outputDirectory);
    final Map<Path, ?> separateFilesContent = getSeparateFilesContent(format, asyncApiSpec, generator, aspectName);

    for (final Map.Entry<Path, ?> entry : separateFilesContent.entrySet()) {
        writeFile(root.resolve(entry.getKey()), entry.getValue());
    }
}

private Map<Path, ?> getSeparateFilesContent(final ApiFormat format, final AsyncApiSchemaArtifact asyncApiSpec, final AspectModelAsyncApiGenerator generator, final String aspectName) {
    final String formatString = format == ApiFormat.JSON ? "json" : "yaml";
    final Map<Path, JsonNode> separateFilesContent = generator.getContentWithSeparateSchemas(asyncApiSpec.getContent(), formatString, aspectName);

    if (format == ApiFormat.JSON) {
        return separateFilesContent;
    }
     
    return separateFilesContent.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry -> jsonToYaml(entry.getValue())));
    }
}

private void writeFile(Path filePath, Object content) throws IOException {
    try (final OutputStream out = new FileOutputStream(filePath.toFile())) {
        if (content instanceof JsonNode) {
            OBJECT_MAPPER.writerWithDefaultPrettyPrinter().writeValue(out, content);
        } else if (content instanceof String) {
            out.write(((String) content).getBytes(StandardCharsets.UTF_8));
        }
    }
}
```

Reply:
Could you please describe what you mean...?

Because this functionality working on "client" side and can be override on like you want? 

Same logic we use for openApi

Reply:
It is just a suggestion to simplify the redundancy in the programme code. However, this does not necessarily have to be refactored.